src/napari_micromanager/main_window.py

```diff
@@ -344,6 +354,23 @@ def _get_defaultdict_layers(self, event: ActiveMDAEvent) -> defaultdict[Any, set
                 layergroups[key].add(lay)
         return layergroups
 
+    def _get_scale_from_sequence(
+        self, sequence: MDASequence, layer: napari.layers.Image
+    ) -> Tuple[float, ...]:  # noqa: U006
+        """Calculate and return the layer scale.
+
+        ...using pixel size, layer shape and the MDASequence z info.
+        """
+        scale = [self._mmc.getPixelSizeUm()] * len(layer.data.shape)
+        if len(sequence.z_plan) > 1 and isinstance(
+            sequence.z_plan, (ZTopBottom, ZRangeAround, ZAboveBelow)
+        ):
+            if sequence.axis_order.index("z") == 3:
+                scale[-3] = sequence.z_plan.step
+            elif sequence.axis_order.index("z") == 2:
+                scale[-4] = sequence.z_plan.step
+        return tuple(scale)
```

Comment:
couple thoughts:

- `scale = [self._mmc.getPixelSizeUm()] * len(layer.data.shape)` only makes sense if there will _only_ ever be 2 or 3 dimensions to `layer.data`, otherwise you're also scaling the time dimension, etc...  `use [1] * len()` and then only set the last two to the pixel size
- avoid using `list.index('something')` twice in the same block.  Just index it once and use it twice
- i think you want `sequence.used_axes` here instead of axis_order.
so, something like this?

```suggestion
        scale = [1] * len(layer.data.shape)
        scale[-2:] = [self._mmc.getPixelSizeUm()] * 2
        try:
            index = sequence.used_axes.index('z')
            scale[index] = getattr(sequence.z_plan, 'step', 1)
        except ValueError:
            pass # z not in used_axes afterall ?
        return tuple(scale)
```

Reply:
however we need to differentiate when we split the channels...otherwise the scale is not correct.

for example, if we have 2 channels and a z stack with step size 2µm, we get:
- scale = [1. 1. 2. 1.] with layer shape = (1, 3, 512, 512) 
when actually should be:
- scale = [1. 2. 1. 1.] with layer shape = (1, 3, 512, 512) 

that's why I used `sequence.axis_order.index("z")` and changing the index like below:
```
 ...
 index = sequence.axis_order.index("z")
        if index == 2:
                index = -4
        elif index == 3:
                index = -3
...
```
What do you think?

Reply:
I'll write a proper test soon